TestResultSummaryService/parsers/Parser.js

```diff
@@ -9,7 +9,7 @@ class Parser {
 
     exactJavaVersion(output) {
         const javaVersionRegex = /=JAVA VERSION OUTPUT BEGIN=[\r\n]+([\s\S]*?)[\r\n]+.*=JAVA VERSION OUTPUT END=/;
-        const javaBuildDateRegex = /\s([0-9]{4})-?(0[1-9]|1[012])-?(0[1-9]|[12][0-9]|3[01])/;
+        const javaBuildDateRegex = /([0-9]{4})-?(0[1-9]|1[012])-?(0[1-9]|[12][0-9]|3[01])/;

```

Comment:
I think this code should be consistent. Instead of looking for both `openjdk version` date and `build ` date, we should only look for `build`, not `openjdk version`. 

As stated in https://github.com/adoptium/aqa-test-tools/issues/360#issuecomment-803686236, we may not have a date in `openjdk version`. Even if we have a date in `openjdk version`, it maybe future a release date.

Reply:
As discussed, we should look for build date first. If it does not exist, then look for openjdk version date